### DISCUSSION TITLE
Is it correct for `regrid_to_lower_res()` to assume second axis is always latitude?

### DISCUSSION THREAD

**Discussion by tomvothecoder** [E3SM-Project/e3sm_diags#706] [Q&A] [answered]
I'm refactoring this function in PR #677.  It compares the number of latitude coordinates to determine which variable is lower 
resolution.

https://github.com/E3SM-Project/e3sm_diags/blob/d0aa16b099db10159d0b585b24f8c641c5bb780c/e3sm_diags/driver/utils/general.py#L240-L248

**I noticed that this function assumes the second element in `.getAxisList()` is latitude. However, I'm using a test dataset that has a Z axis, `lev` which is the second axis (example below).** **Is `regrid_to_lower_res()` incorrectly comparing the length of `lev`  rather than `lat` when the dataset has a Z axis?**

```python
import cdms2

filepath =  "qa/658-lat-lon-set/cl_Amon_E3SM-2-0_historical_r1i1p1f1_gr_200001-201412.nc"
ds = cdms2.open(filepath)

cl = ds["cl"]

axes1 = cl.getAxisList()

axes1[1]

#    id: lev
#    Designated a level axis.
#    units:  hybrid_sigma_pressure
#    Length: 30
#    First:  3.6434658
#    Last:   992.5561
#    Other axis attributes:
#       axis: Z
#       long_name: hybrid level at layer midpoints (1000*(A+B))
#       positive: down
#       A_var: hyam
#       B_var: hybm
#       P0_var: P0
#       PS_var: PS
#       realtopology: linear
#    Python id:  0x7f58f03254b0
```

### Additional Information

The call stack:

https://github.com/E3SM-Project/e3sm_diags/blob/d0aa16b099db10159d0b585b24f8c641c5bb780c/e3sm_diags/driver/lat_lon_driver.py#L216-L218
https://github.com/E3SM-Project/e3sm_diags/blob/d0aa16b099db10159d0b585b24f8c641c5bb780c/e3sm_diags/driver/lat_lon_driver.py#L21-L32

**Comment by tomvothecoder** [comment]
Do you have any insight @chengzhuzhang?

**Reply by chengzhuzhang** [reply, marked as answer]
regrid_to_lower_res() is always used with assumption that two 2D variables are being compared. In the case of 3D variable is supplied, to extract one vertical slice is an operation as prerequisite. When refactoring, we should use more specific axis, rather than asume second axis is always latitude. 
i.e. it seems like using cmorized dataset for testing revealed a couple weak points, which is good to have. Existing tests mostly use native-ish E3SM output..

**Reply by tomvothecoder** [reply]
Gotcha, I now understand that this function assumes the variables are 2D. 

> In the case of 3D variable is supplied, to extract one vertical slice is an operation as prerequisite. 

I now see the code for subsetting here:

https://github.com/E3SM-Project/e3sm_diags/blob/d0aa16b099db10159d0b585b24f8c641c5bb780c/e3sm_diags/driver/lat_lon_driver.py#L165-L183

> When refactoring, we should use more specific axis, rather than asume second axis is always latitude.

I used the handy `lat = xcdat.get_dim_coords(dataarray, axis="Y")` for this!

**Reply by chengzhuzhang** [reply]
> I used the handy lat = xcdat.get_dim_coords(dataarray, axis="Y") for this!
> 
This is much more elegant!
